transformer_nuggets/quant/nf4_tensor.py

```diff
@@ -1,13 +1,303 @@
 import logging
 from dataclasses import dataclass
-from typing import Dict, Tuple
-
+from typing import Dict, Tuple, Any, Optional, Union
+import math
 import torch
 
 logging.basicConfig(level=logging.INFO)
 
 bnb_available = False
 
+aten = torch.ops.aten
+NF4_OPS_TABLE: Dict[Any, Any] = {}
+
+def implements(aten_ops):
+    """Register aten ops to the float8 op table"""
+
+    def decorator(func):
+        for op in aten_ops:
+            NF4_OPS_TABLE[op] = func
+        return func
+
+    return decorator
+
+@implements(
+    [
+        aten.detach.default,
+    ]
+)
+def nf4_detach(aten_op, args, kwargs=None):
+    # nn.Parameter need detach
+    quantized_data = aten_op(args[0].quantized_data, *args[1:], **kwargs)
+    tensor_meta = SubclassTensorArgs(
```

Comment:
https://github.com/drisspg/transformer_nuggets/pull/24/files#diff-6954806b06a395823616c5a0fefc77b5cf274b3f88a5381c38856cbf74d6f2bfR37-R45

I hijacked the tensor unflatten/flatten to construct the new tensors,